### PR TITLE
Refactor http api and drop spring dependency

### PR DESCRIPTION
**Pull request by jhorstmann**

### CODE REVIEW

File: fahrschein-http-api/src/main/java/org/zalando/fahrschein/http/api/HeadersImpl.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+            }
+        } else {
+            return readOnly ? Collections.unmodifiableList(list) : list;
+
+        }
+    }
+
+    @Override
+    public void add(String headerName, String value) {
+        get(headerName).add(value);
```
Comment (lukasniemeier-zalando) [inline fahrschein-http-api/src/main/java/org/zalando/fahrschein/http/api/HeadersImpl.java]: Why can I add stuff if it is `readOnly`

**Comment by lukasniemeier-zalando** [comment]
:+1:

File: fahrschein-http-api/src/main/java/org/zalando/fahrschein/http/api/HeadersImpl.java (hunk 1 above)
Comment (jhorstmann) [inline fahrschein-http-api/src/main/java/org/zalando/fahrschein/http/api/HeadersImpl.java]: I'll add some more testcases, but the returned list should be unmodifiable.

Comment (jhorstmann) [inline fahrschein-http-api/src/main/java/org/zalando/fahrschein/http/api/HeadersImpl.java]: And of course there was a bug in exactly this logic

**Comment by lukasniemeier-zalando** [comment]
:+1:

**Comment by jhorstmann** [comment]
:+1:

**Pull request merged**